### ISSUE TITLE
Android Wear support

### ISSUE THREAD

**Issue by marksev1** [Waboodoo/HTTP-Shortcuts#25]
Would be super-cool to be able to toggle http post requests from an android wear watch. Especially for IOT applications. What do you think?

**Comment by Waboodoo** [comment]
I agree that this would be an awesome addition, and you are not the first person to bring this suggestion to me. I will see how much effort it would be to add wearable support, though with my current time-management I can make no promises.

**Comment by marksev1** [comment]
Cool :)

**Comment by roccomuso** [comment]
In my spare time I've been doing something similar to http-shortcuts, then I saw this project sharing my smae thoughts. :) Anyway +1 for the wear support.

**Comment by marksev1** [comment]
Any news regarding this support, roccomuso have you done your app?

**Comment by roccomuso** [comment]
#marksev1 You mean the one running on the smartwatch? Not yet. Pretty busy in this period :/ but we should consider a branch of this project, for the android watch. I'd buy a watch just for this :)

**Comment by Waboodoo** [comment]
I don't have a watch myself and also don't have access to one anywhere, so this one is quite difficult to do for me. Thus, unfortunately, I currently cannot work on this. Should I get my hands on one, however, plus a bit of spare time, I will look into it. As always, no promises.

**Comment by marksev1** [comment]
Is emulator not sufficient?

**Comment by Waboodoo** [comment]
That is true, for most of the development the emulator would suffice. But I like to try out my work myself ;).

**Comment by Kusig** [comment]
+1 more vote for this.

**Comment by stefxx** [comment]
4 years later... Any update? I would love a Wear OS version!

**Comment by Waboodoo** [comment]
Unfortunately, I don't really have an update here. I have looked into this at some point and come to the conclusion that it would be significantly more work to develop, QA and publish such a feature than I am willing or able to put into this at the current time. On top of that, I still don't have a smartwatch myself so I would be limited in testing this and in evaluating whether it provides the value and quality that I would like.
I completely agree that it would be a great addition to the app and would have a lot of good uses, but it won't happen in the near future unless I receive help from someone on this topic.

**Comment by stefxx** [comment]
Thanks. Disappointed, but fully understandable. Keep up the good work!

**Comment by Percy2Live** [comment]
+1 would be lit to see this feature

**Comment by mguinness** [comment]
Although most people would prefer a standalone app, I think a simpler solution would be to create a companion app that uses [MessageClient API](https://developers.google.com/android/reference/com/google/android/gms/wearable/MessageClient) to communicate with the phone via Bluetooth.

That way you could develop a simple WearOS app to request the list of shortcuts from the phone app and list them on the watch app.  Pressing an item would then send a message to the phone to trigger and execute the shortcut and return the resulting payload back to the watch for display.

See [From Wrist to Hand: Develop a Companion App for Your Wearable Application](https://developer.samsung.com/sdp/blog/en/2023/10/31/from-wrist-to-hand-develop-a-companion-app-for-your-wearable-application) for details.  I think both apps need to have the same package name to communicate with each other, see [WearableListenerService onMessageReceived is not called on device](https://stackoverflow.com/a/50995336/310601).

I also found another app that does something similar (triggering actions from watch) called [WakeOnLan](https://github.com/Florianisme/WakeOnLan).

**Comment by cmader** [comment]
For sending HTTP GET requests to a server from the watch, I wrote the HttpClient app for Garmin and WearOS devices. Maybe this is also useful for you as it serves my use cases fine. See https://apps.garmin.com/apps/da241207-e929-4cdf-9662-11ab17ffd70d and https://github.com/cmader/HttpClient-WearOS
From a feature point of view, this is of course not comparable to HTTP-Shortcuts

**Comment by mguinness** [comment]
@cmader I sideloaded your [HttpClient APK](https://github.com/cmader/HttpClient-WearOS/blob/main/httpclient-1_0_1.apk) and it worked well for my purposes, so thanks for your efforts developing this.

For others that have a similar need you will likely need a proxy to make requests and process responses.  Below is an example using [Minimal APIs](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis) in ASP.NET Core,

```C#
var builder = WebApplication.CreateBuilder(args);
builder.Services.AddHttpClient();
var app = builder.Build();

app.MapGet("/endpoints", () => Results.File("config.txt", "text/plain"));

app.MapGet("/temp", async (IHttpClientFactory httpClientFactory) =>
{
    var client = httpClientFactory.CreateClient();
    var creds = Convert.ToBase64String(Encoding.ASCII.GetBytes("user:pass"));
    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", creds);
    var result = await client.GetStringAsync("https://example.org/sensors/thermostat");
    return $"Temperature is {result}°C";
});

app.Run();
```
